### PR TITLE
Remove items from paging Model

### PR DESCRIPTION
**Pull request by jgrund**
The paging model currently owns the items it does pagination over.
This ends up being inflexible if we want to page over items of different
types.

  - Update paging to not own items.
  - Implement paging components for dropdwon, update paging component to
be reusable.
  - Implement dropdown component.
  - Write tests for paging component.

Signed-off-by: Joe Grund <jgrund@whamcloud.io>

---
This change is [Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/1443)

### CODE REVIEW

File: iml-gui/crate/src/components/paging.rs [hunk 1]
```diff
@@ -146,60 +144,156 @@ pub fn page_selection_view<T: Debug + Eq + Hash>(_open: bool, rows: usize) -> No
...
+
+#[cfg(test)]
+mod tests {
+    use super::Model;
+
+    #[test]
+    fn test_paging_less() {
+        let mut pager = Model::new(20);
+
+        assert_eq!(pager.has_less(), false);
```
Comment (ip1981) [inline iml-gui/crate/src/components/paging.rs]: ?
```suggestion
        assert!(!pager.has_less());
```

Comment (ip1981) [reply iml-gui/crate/src/components/paging.rs]: There are few other places like this :)

**Comment by jgrund** [comment]
___
*[iml-gui/crate/src/components/tree.rs, line 305 at r1](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/1443#-Ly4NngzBYj8H3uj3Lll:-Ly4fesYCeHwnxWcQGqo:b-k1fiku) ([raw file](https://github.com/whamcloud/integrated-manager-for-lustre/blob/518f68b24924fda1ef7cb453e0cad85908691599/iml-gui/crate/src/components/tree.rs#L305)):*
<details><summary><i>Previously, mkpankov (Michael Pankov) wrote…</i></summary><blockquote>

> ```
>             let tree_node = model.get_mut(&addr)?;
> 
>             tree_node.items.remove(&id);
>             tree_node.paging.total -= 1;
> ```
Should we refactor this to remove an item from specified node by id? I.e.

```rust
remove_item_from_node(model: &mut Model, addr: &Address, id: ?)
```
</blockquote></details>

I'll add a method to the model for this.

**Comment by jgrund** [comment]
___
*[iml-gui/crate/src/components/paging.rs, line 232 at r1](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/1443#-Ly14m0DroqfKVyMyHwW-r1-232:-Ly4fqUk9MSp6xlMbN1-:b13b2c) ([raw file](https://github.com/whamcloud/integrated-manager-for-lustre/blob/518f68b24924fda1ef7cb453e0cad85908691599/iml-gui/crate/src/components/paging.rs#L232)):*
<details><summary><i>Previously, ip1981 (Igor Pashev) wrote…</i></summary><blockquote>

There are few other places like this :)
</blockquote></details>

Ok... I'll update the rest :)

File: iml-gui/crate/src/components/dropdown.rs [hunk 2]
```diff
@@ -0,0 +1,57 @@
+use crate::{components::Placement, generated::css_classes::C};
```
Comment (johnsonw) [inline iml-gui/crate/src/components/dropdown.rs]: copyright?

Comment (jgrund) [reply iml-gui/crate/src/components/dropdown.rs]: Not yet. Will add to all files before publishing first version.

**Pull request merged**